lib/Bindings/Python/circt/dialects/_seq_ops_ext.py

```diff
@@ -0,0 +1,38 @@
+from mlir.ir import OpView, StringAttr
+
+
+class CompRegOp:
+
+  def __init__(self,
+               data,
+               input,
+               clk,
+               reset,
+               resetValue,
+               *,
+               name=None,
+               loc=None,
+               ip=None):
+    operands = []
+    results = []
+    attributes = {}
+    results.append(data)
+    operands.append(input)
+    operands.append(clk)
+    if reset is not None:
+      operands.append(reset)
+    if resetValue is not None:
+      operands.append(resetValue)
+    if name is not None:
+      attributes["name"] = StringAttr.get(name)
+
+    OpView.__init__(
```

Comment:
Why was this necessary instead of just calling `super().__init__()`?

Reply:
This is the recommended pattern: https://mlir.llvm.org/docs/Bindings/Python/#extending-generated-op-classes

Reply:
ohh... Is `OpView` the super class type?

Reply:
The super-duper class. The super class is the generated one, which this is mixed into, meaning `super().__init__()` is not going to work right.

Reply:
I must not understand the mixin mechanics and/or `super()` semantics. I'm wondering how it is that https://github.com/llvm/circt/blob/faa0e1827a116292156ff32260a195b1caf3cdce/lib/Bindings/Python/circt/dialects/_rtl_ops_ext.py#L52 works.

Reply:
> I must not understand the mixin mechanics and/or `super()` semantics. I'm wondering how it is that
> 
> https://github.com/llvm/circt/blob/faa0e1827a116292156ff32260a195b1caf3cdce/lib/Bindings/Python/circt/dialects/_rtl_ops_ext.py#L52
> 
> works.

Python's class model is fairly different from most things, but people can usually ignore the weird parts if not using various features for mechanisms like this. Key thing is that `super` always resolves to the next entry in the __mro__, relative to this one. The __mro__ is assembled as classes are defined and method dispatch is cooperative. Evaluate `MyClass.__mro__` to see how it is constructed for this hierarchy.

The other load bearing thing is that there is only one metaclass for a hierarchy, and for pybind classes, it brings its own, which always directly handle construction of the  basemost, native classes. Practically, this means that one and only one subclass in the hierarchy must call __init__ of the outermost pybind class (in this case, always OpView).

It is pretty weird, and this definitely qualifies as an advanced usage. I could have simplified it somewhat if not for the pybind interop, which introduces a number of practical constraints that you don't get with normal python class hierarchies.

Reply:
Remember that the mro is established for the instance, based on the class constructed. Super classes will slot in there somewhere, but the total order can be different than the declared order of inheritance declarations (and is different in this case).